src/main/java/core/basesyntax/utility/impl/ListServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.utility.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.utility.ListService;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ListServiceImpl implements ListService {
+    @Override
+    public Map<String, Integer> getComputedMap(List<FruitTransaction> fruitTransactionList) {
+        Map<String, Integer> fruitTransactionComputed = new HashMap<>();
+        for (FruitTransaction fruitTransaction : fruitTransactionList) {
+            if (!fruitTransactionComputed.containsKey(fruitTransaction.getFruit())) {
```

Comment:
You don't need to check if the key exists twice. Instead, you can streamline the code using the merge method provided by the Map interface. 'merge' will either add a new entry or update an existing entry in a single line. This method takes care of checking the existence of the key and updating the value appropriately.
`for (FruitTransaction fruitTransaction : fruitTransactionList) {
            fruitTransactionComputed.merge(fruitTransaction.getFruit(), 
                                           fruitTransaction.getQuantity(), 
                                           Integer::sum);
        }`